code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -291,11 +297,15 @@ All foods are distributed among various categories. Use common sense.
 			else
 				S.reagents.add_reagent(r_id, amount)
 
-/obj/item/reagent_containers/food/snacks/microwave_act(obj/machinery/microwave/M)
+/obj/item/reagent_containers/food/snacks/microwave_act(obj/machinery/microwave/M, var/quality_increase)
 	var/turf/T = get_turf(src)
 	var/obj/item/result
 	if(cooked_type)
 		result = new cooked_type(T)
+		//if the result is food, set its food quality to the original food item's quality
+		if(istype(result, /obj/item/reagent_containers/food))
```

Comment:
You should add an `isfood` helper define (see __DEFINES/is_helpers.dm) at this point.

Reply:
done